### PR TITLE
Add serviceAccountToken method.

### PR DESCRIPTION
**Pull request by MancunianSam**
This will call the auth server and get a service account token for a
client id and secret.

### CODE REVIEW

File: project/Dependencies.scala [hunk 1]
```diff
@@ -9,4 +9,12 @@ object Dependencies {
   lazy val jbossLogging = "org.jboss.logging" % "jboss-logging" % "3.4.1.Final"
   lazy val keycloakMock = "com.tngtech.keycloakmock" % "mock" % "0.2.0"
   lazy val oauth2 = "com.nimbusds" % "oauth2-oidc-sdk" % "7.1.1"
+  lazy val sttp = "com.softwaremill.sttp.client" %% "core" % "2.1.1"
+  lazy val sttpCirce = "com.softwaremill.sttp.client" %% "circe" % "2.1.1"
+  lazy val sttpAsyncClient = "com.softwaremill.sttp.client" %% "async-http-client-backend-future" % "2.0.0-RC9"
+  lazy val circeCore = "io.circe" %% "circe-core" % "0.13.0"
+  lazy val circeGeneric = "io.circe" %% "circe-generic" % "0.13.0"
+
```
Comment (TomJKing) [inline project/Dependencies.scala:17]: Unnecessary line breaks

File: src/main/scala/uk/gov/nationalarchives/tdr/keycloak/KeycloakUtils.scala [hunk 2]
```diff
@@ -28,8 +37,32 @@ class KeycloakUtils(url: String) {
...
+      .body(body)
+      .auth.basic(clientId, clientSecret)
+      .post(uri"$url/realms/tdr/protocol/openid-connect/token")
+      .response(asJson[AuthResponse])
+      .send()
+
+    val authResponse = response.flatMap { r =>
+      r.body match {
+        case Right(body) => Future.successful(body)
+        case Left(_) => Future.failed(HttpError("An error occurred contacting the auth server"))
```
Comment (suzannehamilton) [inline src/main/scala/uk/gov/nationalarchives/tdr/keycloak/KeycloakUtils.scala:56]: Is there any useful information inside the `Left` that we should include in the error for debugging purposes?

Comment (MancunianSam) [reply src/main/scala/uk/gov/nationalarchives/tdr/keycloak/KeycloakUtils.scala]: There's not loads of information. The message on the exception is null but there is some information in the exception body which could help us track things. I've changed it to return the exception.

File: project/Dependencies.scala (hunk 1 above)
Comment (MancunianSam) [reply project/Dependencies.scala]: I've removed the line breaks.

**Pull request merged**